arangod/Agency/State.cpp

```diff
@@ -1247,11 +1234,11 @@ bool State::loadRemaining(index_t cind) {
 
 /// Find entry by index and term
 bool State::find(index_t prevIndex, term_t prevTerm) {
-  MUTEX_LOCKER(mutexLocker, _logLock);
-  if (prevIndex > _log.size()) {
+  std::lock_guard mutexLocker{_logLock};
+  if (prevIndex >= _log.size()) {
```

Comment:
this fix should be backported to stable versions.

Reply:
Yes, here